File: samples/Windows/java/TableManagerSamples/PrintTableListener/src/EachRowListener.java
package printtablelistener;

import java.text.SimpleDateFormat;
import java.util.Calendar;

import com.fxcore2.*;

public class EachRowListener implements IO2GEachRowListener {
    private SimpleDateFormat mDateFormat = new SimpleDateFormat("MM.dd.yyyy HH:mm:ss");
    private String mAccountID;

    public EachRowListener(String sAccountID) {
        mAccountID = sAccountID;
    }

    @Override
    public void onEachRow(String sRowID, O2GRow rowData) {
        if (rowData.getTableType() == O2GTableType.ORDERS) {
            int columnsCount = rowData.getColumns().size();
            for (int i = 0; i < columnsCount; i++) {
                if (mAccountID.isEmpty() || mAccountID.equals(((O2GOrderTableRow)rowData).getAccountID())) {
                    System.out.print(rowData.getColumns().get(i).getId() + "=");
                    if (Calendar.class.isAssignableFrom(rowData.getCell(i).getClass())) {
                        System.out.print(mDateFormat.format(((Calendar)rowData.getCell(i)).getTime()));
                    } else {
                        System.out.print(rowData.getCell(i));
                    }
                    System.out.print("; ");
                }
            }
            System.out.println("");
        }
    }
}

File: samples/Linux/netcore_for_Linux_x64_only/netcore1.0/TableManagerSamples/PrintTable/src/EachRowListener.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using fxcore2;

namespace PrintTable
{
    class EachRowListener : IO2GEachRowListener
    {
        private string mAccountID;

        public EachRowListener(string sAccountID)
        {
            mAccountID = sAccountID;
        }

        public void onEachRow(string sRowID, O2GRow rowData)
        {
            if (rowData.TableType == O2GTableType.Orders)
            {
                int columnsCount = rowData.Columns.Count;
                for (int i = 0; i < columnsCount; i++)
                {
                    if (string.IsNullOrEmpty(mAccountID) || mAccountID.Equals(((O2GOrderTableRow)rowData).AccountID))
                    {
                        Console.Write(rowData.Columns[i].ID + "=" + rowData.getCell(i) + "; ");
                    }
                }
                Console.WriteLine("");
            }
        }
    }
}
